Testing with Qt6 misses dialog popups

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

When running tests with Qt6 some dialogs seem to be missed and thus aren't clicked on. For example, the PartDesignTransformed test from TestPartDesignGui will hang as the second dialog to select the body will not be closed.

```
Testing moving one feature from one body to another
PartDesign::Body: Link(s) to object(s) 'Pad' go out of the allowed scope 'Body'. Instead, the linked object(s) reside within 'Body001 Body001'.
Testing refuse to move the feature with dependencies from one body to another
Testing applying MultiTransform to the Box outside the body
Traceback (most recent call last):
  File "/home/jookia/FreeCAD/build_qt6_reldeb/Mod/PartDesign/TestPartDesignGui.py", line 56, in __call__
    self.test.assertIsNone(diag, "Dialog box could not be found")
  File "/usr/lib/python3.12/unittest/case.py", line 1281, in assertIsNone
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python3.12/unittest/case.py", line 715, in fail
    raise self.failureException(msg)
AssertionError: <PySide6.QtWidgets.QDialog(0x7ffdb0dd6050, name="PartDesignGui__DlgActiveBody") at 0x74ac66342340> is not None : Dialog box could not be found
```

### Full version info

```shell
OS: Arch Linux
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37842 (Git)
Build type: Release
Branch: jookia_testing
Hash: 0f9dcefb3f8a99969117ff60167e6a8fdb26706d
Python 3.12.3, Qt 6.7.1, Coin 4.0.2, Vtk 9.3.0, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * fasteners 0.4.67

(Note: based on 9b547bb90fbdf5f9e6df17fee640566cbe2ab37e)
```


### Subproject(s) affected?

None

### Anything else?

Related to https://github.com/FreeCAD/FreeCAD/issues/13303

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/Mod/PartDesign/TestPartDesignGui.py
+++ b/src/Mod/PartDesign/TestPartDesignGui.py
@@ -39,42 +39,40 @@ class CallableCheckWorkflow:
     def __init__(self, test):
         self.test = test
     def __call__(self):
-        diag = QApplication.activeModalWidget()
-        self.test.assertIsNotNone(diag, "Dialog box could not be found")
-        if (diag is not None):
-            cbuttons = diag.findChildren(QtGui.QPushButton)
-            cbutton = cbuttons[1]
-            dialogcheck = CallableCheckDialog(self.test)
+        dialog = QApplication.activeModalWidget()
+        self.test.assertIsNotNone(dialog, "Dialog box could not be found")
+        if (dialog is not None):
+            dialogcheck = CallableCheckDialogWasClosed(self.test)
             QtCore.QTimer.singleShot(500, dialogcheck)
-            QtCore.QTimer.singleShot(0, cbutton, QtCore.SLOT('click()'))
+            QtCore.QTimer.singleShot(0, dialog, QtCore.SLOT('accept()'))
 
-class CallableCheckDialog:
+class CallableCheckDialogWasClosed:
     def __init__(self, test):
         self.test = test
     def __call__(self):
-        diag = QApplication.activeModalWidget()
-        self.test.assertIsNone(diag, "Dialog box could not be found")
+        dialog = QApplication.activeModalWidget()
+        self.test.assertIsNone(dialog, "Dialog box was not closed by accept()")
 
 class CallableCheckWarning:
     def __init__(self, test):
         self.test = test
     def __call__(self):
-        diag = QApplication.activeModalWidget()
-        self.test.assertIsNotNone(diag, "Input dialog box could not be found")
-        if (diag is not None):
-            QtCore.QTimer.singleShot(0, diag, QtCore.SLOT('accept()'))
+        dialog = QApplication.activeModalWidget()
+        self.test.assertIsNotNone(dialog, "Input dialog box could not be found")
+        if (dialog is not None):
+            QtCore.QTimer.singleShot(0, dialog, QtCore.SLOT('accept()'))
 
 class CallableComboBox:
     def __init__(self, test):
         self.test = test
     def __call__(self):
-        diag = QApplication.activeModalWidget()
-        self.test.assertIsNotNone(diag, "Warning dialog box could not be found")
-        if (diag is not None):
-            cbox = diag.findChild(QtGui.QComboBox)
+        dialog = QApplication.activeModalWidget()
+        self.test.assertIsNotNone(dialog, "Warning dialog box could not be found")
+        if (dialog is not None):
+            cbox = dialog.findChild(QtGui.QComboBox)
             self.test.assertIsNotNone(cbox, "ComboBox widget could not be found")
             if (cbox is not None):
-                QtCore.QTimer.singleShot(0, diag, QtCore.SLOT('accept()'))
+                QtCore.QTimer.singleShot(0, dialog, QtCore.SLOT('accept()'))
 
 App = FreeCAD
 Gui = FreeCADGui

